diff -up kdelibs-3.5.10/kio/kio/configure.in.in.types kdelibs-3.5.10/kio/kio/configure.in.in
--- kdelibs-3.5.10/kio/kio/configure.in.in.types	2006-07-22 10:16:37.000000000 +0200
+++ kdelibs-3.5.10/kio/kio/configure.in.in	2010-07-20 13:02:01.000000000 +0200
@@ -122,17 +122,14 @@ if test "x$kde_enable_inotify" = "xyes";
 
   AC_TRY_COMPILE(
   [
-#include <asm/unistd.h>
-#define _S390_BITOPS_H
-#include <linux/inotify.h>
+#include <unistd.h>
+#include <sys/inotify.h>
   ],
   [
 #ifndef IN_ALL_EVENTS
 #error no inotify notification
 #endif
-#ifndef __NR_inotify_init
-#error no __NR_inotify_init
-#endif
+inotify_init();
 
   ],kde_cv_have_inotify=yes,kde_cv_have_inotify=no)
 
diff -up kdelibs-3.5.10/kio/kio/kdirwatch.cpp.types kdelibs-3.5.10/kio/kio/kdirwatch.cpp
--- kdelibs-3.5.10/kio/kio/kdirwatch.cpp.types	2010-07-20 11:19:56.000000000 +0200
+++ kdelibs-3.5.10/kio/kio/kdirwatch.cpp	2010-07-20 12:23:09.000000000 +0200
@@ -73,10 +73,11 @@
 #define _S390_BITOPS_H
 #define flock linux_flock
 #define flock64 linux_flock64
-#include <linux/inotify.h>
+#include <sys/inotify.h>
 #undef flock
 #undef flock64
 
+/*
 static inline int inotify_init (void)
 {
   return syscall (__NR_inotify_init);
@@ -91,7 +92,7 @@ static inline int inotify_rm_watch (int 
 {
   return syscall (__NR_inotify_rm_watch, fd, wd);
 }
-
+*/
 #ifndef  IN_ONLYDIR
 #define  IN_ONLYDIR 0x01000000 
 #endif
